### PR TITLE
Dedicated variable for IPv4 & IPv6

### PR DESCRIPTION
**Pull request by fodhelper**

### CODE REVIEW

**Comment by M03ED** [comment]
SERVER_IPV4 variable its not useful because its duplicate of SERVER_IP variable

**Comment by fodhelper** [comment]
If server has no IPv4, it will take 10 sec to find the correct IPv6
It's admin choice to change
```
def get_public_ip():
    resp = get_public_ipv4()
    if resp == '127.0.0.1':
        return get_public_ipv6()
    else:
        return resp
```
to
```
def get_public_ip():
    try:
        resp = requests.get('http://api4.ipify.org/', timeout=5).text.strip()
        if ipaddress.IPv4Address(resp).is_global:
            return resp
    except:
        pass

    try:
        resp = requests.get('http://icanhazip.com/', timeout=5).text.strip()
        if ipaddress.IPv4Address(resp).is_global:
            return resp
        elif ipaddress.IPv6Address(resp).is_global:
            return '[' + resp + ']'
    except:
        pass

    return '127.0.0.1'
```

Can also change the first request timeout to 3 to fast switch to the second api if the first one failed

**Comment by fodhelper** [comment]
> SERVER_IPV4 variable its not useful because its duplicate of SERVER_IP variable

I thought about the same thing, but each variable has it's own advantages 
Do you want me to remove it?

**Comment by fodhelper** [comment]
The point is
if you use SERVER_IPV4 and have no IPv4, there is no fallback to ipv6
If you use SERVER_IPV6 have no IPv6, there is no fallback to IPv4
But SERVER_IP checks IPv4 first, and then fallbacks to IPv46

**Comment by M03ED** [comment]
lets see what @SaintShit think about this

File: app/utils/system.py [hunk 1]
```diff
@@ -108,33 +108,45 @@ def check_port(port: int) -> bool:
...
+    except:
         pass
-    finally:
-        sock.close()
 
-    return '127.0.0.1'
+    try:
+        resp = requests.get('http://ipv6.icanhazip.com/', timeout=5).text.strip()
+        if ipaddress.IPv6Address(resp).is_global:
+            return '[' + resp + ']'
```
[the remark this answers is not in the document]
Comment (fodhelper) [reply app/utils/system.py]: is it correct now?

[the remark this answers is not in the document]
Comment (M03ED) [reply app/utils/system.py]: This is lazy mode and it's same as fstring and it's correct

**Comment by mikeesierrah** [comment]
> > SERVER_IPV4 variable its not useful because its duplicate of SERVER_IP variable
> 
> I thought about the same thing, but each variable has it's own advantages
> 
> if you use SERVER_IPV4 and have no IPv4, there is no fallback to ipv6 (but there is 2 api to check for ipv4) If you use SERVER_IPV6 have no IPv6, there is no fallback to IPv4 (but there is 2 api to check for ipv6) But SERVER_IP checks IPv4 first, and then fallbacks to IPv46
> 
> Do you want me to remove it?
He's got a point

**Comment by fodhelper** [comment]
I think SERVER_IP(IPv4 only) and SERVER_IPV6 is enough, i will remove SERVER_IPV4 later
If a user's server have no IPv4 (rare case), user must use SERVER_IPV6 then

**Comment by SaintShit** [comment]
To contribute to the project, please clone it and test the changes on your local machine. it seems you used UI editor that makes lots of commits. and also, there might be some bugs as code has not runned after changes. for example, function `get_public_ipv6` isn't imported in `app/subscription/share.py`

**Pull request closed without merging**